engine/framework/master.go

```diff
@@ -670,6 +682,39 @@ func (m *DefaultBaseMaster) IsMasterReady() bool {
 	return m.workerManager.IsInitialized()
 }
 
+// Exit implements BaseMaster.Exit
+func (m *DefaultBaseMaster) Exit(ctx context.Context, exitReason ExitReason, err error, extMsg string) (errRet error) {
+	// Set the errCenter to prevent user from forgetting to return directly after calling 'Exit'
+	defer func() {
+		errTmp := err
```

Comment:
do you mean

```suggestion
		errTmp := errRet
```

or why we use named return?

Reply:
We can do the `err` => `errRet` here.
`err` is the original error, but not return error here.
I will use a better name to show this attemp.

Reply:
I will simplify the logic